src/odemis/driver/technolution.py

```diff
@@ -368,69 +435,92 @@ def _acquire(self):
         '''
-        self._field_data = FieldMetaData(
-                field_num[0] * self.parent._ebeam_scanner.resolution.value[0],
-                field_num[1] * self.parent._ebeam_scanner.resolution.value[1])
-        #TODO K.K. remove temporary vas A
-        A = self.convert_field_num2pixels(field_num)
-        self.acq_queue.put(("next", self._field_data, self.dataContent.value, self.dataFlow.notify))
+        field_data = FieldMetaData(*self.convert_field_num2pixels(field_num))
+        self.acq_queue.put(("next", field_data, self.dataContent.value, self.data.notify))
 
     def stop_acquisition(self):
-        self.acq_queue.put(("stop", ))
+        """
+        Puts a 'stop' field image scan on the queue, after this call, no fields can be scanned anymore. A new mega
+        field can be started. The call triggers the post prosessing process to generate and offload additional zoom
+        levels
+        """
+        self.acq_queue.put(("stop",))
 
     def acquire_single_field(self, field_num=(0, 0)):
+        """
+        Puts a the series 'start','next','stop' commands on the queue with the appropriate metadata models and
+        scans a single field image. By providing as notifier function a return_queue the image can be returned. The
+        use of the queue allows the use of the timeout functionality
+
+        :param field_num:
+        :return: DA of the single field image
+        """
+
         return_queue = queue.Queue()  # queue which allows to return images and be blocked when waiting on images
         mega_field_data = self._assemble_megafield_metadata()
 
         self.acq_queue.put(("start", mega_field_data))
-        self._field_data = FieldMetaData(field_num[0] * self.parent._ebeam_scanner.resolution.value[0],
-                                         field_num[1] * self.parent._ebeam_scanner.resolution.value[1])
-        self.acq_queue.put(("next", self._field_data, self.dataContent.value, return_queue.put))
-        self.acq_queue.put(("stop", ))
+        field_data = FieldMetaData(*self.convert_field_num2pixels(field_num))
 
-        return return_queue.get(timeout=600)
-
-    def terminate(self):
-        # Clear the queue
-        while True:
-            try:
-                self.acq_queue.get(block=False)
-            except queue.Empty:
-                break
+        self.acq_queue.put(("next", field_data, self.dataContent.value, return_queue.put))
+        self.acq_queue.put(("stop",))
 
-        self.acq_queue.put(("terminate", None))
+        return return_queue.get(timeout=600)
 
     def convert_field_num2pixels(self, field_num):
-        return (field_num[0] * self._shape.value[0] * self.parent._ebeam_scanner.resolution.value[0],
-                field_num[1] * self._shape.value[1] * self.parent._ebeam_scanner.resolution.value[1])
+        return (field_num[0] * self.parent._ebeam_scanner.resolution.value[0],
+                field_num[1] * self.parent._ebeam_scanner.resolution.value[1])
```

Comment:
Can you not define in the init of the component or somewhere else that `scanner = self.parent._ebeam_scanner`?

Reply:
@pieleric do you think witin this object everywhere such a reference should be made? so self.scanner  = self.parent._ebeam_scanner? (and also for self.descanner)
This explicily gives the relation between the children however it is not nice for the eyes.

Reply:
If you need this reference, it's no problem to either store it at init, or even pass it in the `__init__()`. Actually this later idea is probably even better because it makes it explicit that the initialisation should happen after the scanner and descanner have been created.